sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -453,6 +453,12 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    *   format("json").save(path)
    * }}}
    *
+   * You can set the following JSON-specific options for writing JSON files:
+   * <li>`compression` or `codec` (default `null`): compression codec to use when saving to file.
```

Comment:
actually i'd remove codec support from the underlying source code, and only keep it for csv as an undocumented option for backward compatibility.